spec/workers/proactive_project_sync_worker_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec::Matchers.define :have_been_queued_for_project_sync do
+  match do |actual|
+    expect(PackageManagerDownloadWorker).to receive(:perform_async).once.with(actual.sync_classes.first.name, actual.name, any_args)
+    expect(CheckStatusWorker).to receive(:perform_async).once.with(actual.id)
+  end
+end
+
+RSpec::Matchers.define :have_not_been_queued_for_project_sync do
+  match do |actual|
+    expect(PackageManagerDownloadWorker).not_to receive(:perform_async).with(String, actual.name, any_args)
+    expect(CheckStatusWorker).not_to receive(:perform_async).with(actual.id)
+  end
+end
+
```

Comment:
How do we feel about using custom matchers here to make the specs fluent? Easier to follow or nah?

Reply:
Easier to follow, this is a good abstraction. :+1: